### ISSUE TITLE
Broken parts on version 3

### ISSUE THREAD

**Issue by mrbrandao** [ajrnz/rundeck-telegram-plugin#8]
Hi there, I've been using this plugin for a year ago always in version 2.10+, but it seems to have a little broke part in version3. For example, ${dateEnded} doesn't work anymore  returning this exception:

```
rundeck stderr | Can't convert the date to a string, because it is not known which parts of the date variable are in use. Use ?date, ?time or ?datetime built-in, or ?string.<format> or ?string(format) built-in with this date.
The problematic instruction:
==> ${dateEnded} [on line 3, column 1 in message]
```
So far this is the only issue I found from the version2 to the version3. It would be very nice some fix for that... ;-)

**Comment by NFDWADM** [comment]
This happens because rundeck changed their date format for dateEnded.

I was able to fix it in repl but I couldn't get it to compile since I know nothing about scala.

```
    val sourceparser = new SimpleDateFormat("EEE MMM dd HH:mm:ss z yyyy")
    val targetparser = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS")
    val datefix = sourceparser.parse(executionData("dateEnded"))
    executionData(dateEnded) = targetparser.format(datefix)

```

Fails with 
```
[error] /home/user/misc/rundeck-telegram-plugin/src/main/scala/ajr/rundeck/telegram/TelegramNotificationPlugin.scala:216: type mismatch;
[error]  found   : String("dateEnded")
[error]  required: _$15 where type _$15
[error]     val datefix = sourceparser.parse(executionData("dateEnded"))
```

So if anybody with a bit of scala knowledge feels like telling me what I did wrong I'd be happy.

**Comment by ajrnz** [comment]
Hi, I don't have much time to  look at this at the moment but happy to help and do a release.  `executionData` is just a java map so you should just be able to use `put`

    executionData.put("<key>", value)

Rather than replacing the value it would be better to use another key say `dateEndedIsoString` so it remains compatible with the old versions

**Comment by NFDWADM** [comment]
Thank you for your help.

I tried that but I must have the syntax wrong

```
    val sourceparser = new SimpleDateFormat("EEE MMM dd HH:mm:ss z yyyy")
    val targetparser = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS")
    val wrongdate = executionData.get("dateEnded")
    val dateparsed = sourceparser.parse(wrongdate.toString())
    executionData.put("dateEndedIsoString", targetparser.format(dateparsed))
```

sbt fails with
```
[error] /home/sas/misc/rundeck-telegram-plugin/src/main/scala/ajr/rundeck/telegram/TelegramNotificationPlugin.scala:218: type mismatch;
[error]  found   : String("dateEndedIsoString")
[error]  required: _$15
[error]     executionData.put("dateEndedIsoString", targetparser.format(dateparsed))
```

I am guessing it's because executionDate is typed as _,_ but I don't know how to work around that.

**Comment by ajrnz** [comment]
Ah that's scala not liking Java's lack of types.

I've pushed a change to an `iso-date-strings` branch: https://github.com/ajrnz/rundeck-telegram-plugin/tree/iso-date-strings

Can you test it and let me know if it works?

**Comment by NFDWADM** [comment]
I confirm that 

```
${dateEndedIsoString}
${dateStartedIsoString}
```
work as expected, thank you!

I found another problem in the meantime, I'll create a new ticket.

**Comment by ajrnz** [comment]
Fixed by #10
